src/mono/mono/metadata/object.c

```diff
@@ -5681,6 +5681,28 @@ mono_array_new_full_checked (MonoClass *array_class, uintptr_t *lengths, intptr_
 	return array;
 }
 
+MonoArray *
+mono_array_new_2_jagged_checked (MonoClass *klass, uintptr_t length1, uintptr_t length2, MonoError *error)
+{
+	MonoClass *elem_class = m_class_get_element_class (klass);
+	MonoArray *ret;
+
+	g_assert (m_class_get_rank (klass) == 1);
+	g_assert (m_class_get_rank (elem_class));
+
+	// FIXME Handle ?
+	ret = mono_array_new_full_checked (klass, &length1, NULL, error);
+	return_val_if_nok (error, NULL);
```

Comment:
This needs handles because the later alloc can cause a GC.
Do:
	HANDLE_FUNCTION_ENTER ();

MONO_HANDLE_PIN (ret);

and later:
MONO_HANDLE_FUNCTION_RETURN_VAL (ret);